sbol_utilities/package.py

```diff
@@ -94,26 +95,33 @@ def define_package(doc: sbol3.Document):
         # Namespace must match the hasNamespace value for all of its members
         package.namespace = package_namespace
 
+        # Combine all of the docs into one
+        doc = args[0]
+        for sbol_doc in args[1:]:
+            doc_objects = [o for o in sbol_doc.objects]
+            doc.add(doc_objects)
+        
         # Add the package to the document
         doc.add(package)
 
         # Return the doc
         return(doc)
         
 
-def check_namespaces(doc: sbol3.Document):
-    """ Check if the namespaces of all top level objects are the same
+def check_namespaces(*args: sbol3.Document):
```

Comment:
I think we should change this to apply to the package object, rather than the document. The approach we've embraced in pySBOL is "build, then check".

Reply:
I have updated the check_namespaces function to work on a package, it now checks the namespaces of all of the members of the package, checks that all sub-packages have a namespace with a stem of the namespace of the root package, and then checks that each subpackage has members of a single namespace